scene/2d/tile_map_layer.cpp

```diff
@@ -2096,6 +2153,12 @@ void TileMapLayer::_notification(int p_what) {
 			dirty.flags[DIRTY_FLAGS_LAYER_VISIBILITY] = true;
 			_queue_internal_update();
 		} break;
+
+		case NOTIFICATION_DRAW: {
+			if (has_scalable_textures && get_viewport() && (get_viewport()->get_oversampling() != oversampling)) {
```

Comment:
Can you use the `dirty.flags` system here to notify that the oversampling status changed instead ? You can add a flag value in the enum for oversampling.